### ISSUE TITLE
Cannot reproduce 94.72% Top-1 accuracy on CIFAR10 (4000 labeled data)

### ISSUE THREAD

**Issue by tqtrunghnvn** [kekmodel/FixMatch-pytorch#8]
I tried to reproduce this source code on CIFAR10 (4000 labeled data), but I only got ~93% Top-1 accuracy.

Is there anyone meet the same problem?

**Comment by wangtingwei1993** [comment]
It seems that I get a similar accuracy. Did you find what happened?

**Comment by tqtrunghnvn** [comment]
@wangtingwei1993  
Not yet.
Did you just make any changes before starting training your model?

In my case, I just cloned this repository, and then starting training my model without any changes.

**Comment by wangtingwei1993** [comment]
@tqtrunghnvn. I changed nothing.

**Comment by tqtrunghnvn** [comment]
@wangtingwei1993 Did you train your model on single or multiple GPU?

**Comment by wangtingwei1993** [comment]
@tqtrunghnvn Single GPU because. Do you think single or multiple matters?

**Comment by tqtrunghnvn** [comment]
@wangtingwei1993 I am not sure, but I also used a single GPU. 

Could you tell me which commit version did you clone?
I worry that maybe I and you are using different versions.

**Comment by wangtingwei1993** [comment]
@tqtrunghnvn The latest one

**Comment by tqtrunghnvn** [comment]
@wangtingwei1993 I see. Thank you!

**Comment by Fuchai** [comment]
I cannot reproduce the result either. After 1024 epochs, the top-1 accuracy on CIFAR10 4000 converges to around 93% as well.
Torch 1.6.0. Nvidia V100 single GPU. All parameters are default.

**Comment by Chen-Song** [comment]
The top-1 accuracy on CIFAR10 4000 using this code get 93.31%  in my trial as well.

**Comment by wistuba** [comment]
I faced the same problem, obtaining an architecture with about 93%. I ended up using the official implementation of WRN (https://github.com/szagoruyko/wide-residual-networks) and integrated all necessary parts for fixmatch from here. Simply using the default settings for training WRN (which uses only 200 instead of 1k epochs) gave me about 95% accuracy which is already close to the 95.7% reported in the paper. Training for longer with cosine schedule might give that additional lift of 0.7%.
Someone might look into whether the WRN in this repository is implemented correctly.

**Comment by kekmodel** [comment]
> I faced the same problem, obtaining an architecture with about 93%. I ended up using the official implementation of WRN (https://github.com/szagoruyko/wide-residual-networks) and integrated all necessary parts for fixmatch from here. Simply using the default settings for training WRN (which uses only 200 instead of 1k epochs) gave me about 95% accuracy which is already close to the 95.7% reported in the paper. Training for longer with cosine schedule might give that additional lift of 0.7%.
> Someone might look into whether the WRN in this repository is implemented correctly.

I'll check. Thanks.

**Comment by Z1Wu** [comment]
@wistuba I only can get about 94% accuracy after modify the WRN part of this implentation. Actually I can only find a little difference between this implentation and the official repo of WRN((https://github.com/szagoruyko/wide-residual-networks) b ecause I am not very familiar with Lua. Could you please make your modified code public?  Thank you in advance.

**Comment by wistuba** [comment]
@Z1Wu I've used the pytorch code from here: https://github.com/szagoruyko/wide-residual-networks/tree/master/pytorch

**Comment by kekmodel** [comment]
There was a problem with the EMA. Now it reproduces well! Thank you.
